airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt

```diff
@@ -132,14 +141,38 @@ class DefaultDestinationTaskLauncher(
     private val succeeded = Channel<Boolean>(Channel.UNLIMITED)
 
     private val teardownIsEnqueued = AtomicBoolean(false)
+    private val failSyncIsEnqueued = AtomicBoolean(false)
 
-    private suspend fun enqueue(task: LeveledTask) {
-        taskScopeProvider.launch(exceptionHandler.withExceptionHandling(task))
+    inner class TaskWrapper(
```

Comment:
Nit: should this be named something that indicates its purpose is to do error handling?